dbms/tests/queries/0_stateless/01035_logical_functions_with_uint8.sql

```diff
@@ -0,0 +1 @@
+select toUInt8(number) as n from numbers(100) where bitAnd(n, 1) and bitAnd(n, 2) and bitAnd(n, 4);
```

Comment:
Okay, so right now we have a problem with *UInt8* representation of bolean values. I suggest that we add a few more tests  and a comment may be, because this line is far from being self explanatory. It is not clear from the line why we need 3 arguments for `and` in `where` expression and exactly why this is being checked. Probably it'd be nice to have the same checks for constants as well.

Reply:
`select materialize(1) and materialize(2) and materialize(4)`